libpod/container_api.go

```diff
@@ -90,6 +90,14 @@ func (c *Container) Init() (err error) {
 		return errors.Wrapf(err, "unable to copy /etc/hosts to container space")
 	}
 
+	if c.Spec().Hostname == "" {
+		c.config.Spec.Hostname = c.ID()[:12]
```

Comment:
You should check the length of c.ID to make sure it's at least 12 chars long.  I just got bit by this very thing.

Reply:
even if len c.id has to be over 12 ?

Reply:
We should never have an ID that is not 64 characters long. Its generation is entirely internal, and the function for generating them will never error. Bad (uninitialized) input should be caught by validity checks earlier in the function as well.

Reply:
I'm a strong believer in defensive programming.  Even though it should never happen, the segfault is very ugly if it does.  Your call, it's a quick check.

Reply:
fixed

Reply:
@baude Do we ever use the truncated ID in other places? Is it worth having a TruncID() function on containers that gives id[:12] (with optional length check)? I think we use it a bit in the podman CLI, but I don't know if having a function would be useful for that

Reply:
we trunc in ps